File: src/main/java/bg/uniplovdiv/talkify/channel/model/ChannelSearchCriteria.java
package bg.uniplovdiv.talkify.channel.model;

import bg.uniplovdiv.talkify.common.encodedid.EncodedId;

public record ChannelSearchCriteria(
    String name,
    @EncodedId Long userId,
    @EncodedId Long ownerId,
    @EncodedId Long adminId,
    @EncodedId Long guestId,
    Boolean active) {}

File: src/main/frontend/src/models/channel/ChannelSearchCriteria.ts
import Paginated from "../common/Paginated";

interface ChannelSearchCriteria extends Paginated {
  name?: string;
  userId?: string;
  ownerId?: string;
  adminId?: string;
  guestId?: string;
  active?: boolean;
}

export default ChannelSearchCriteria;
